shared_model/builders/protobuf/unsigned_proto.hpp

```diff
@@ -49,14 +49,11 @@ namespace shared_model {
        * @return signed object
        */
       T signAndAddSignature(const crypto::Keypair &keypair) {
-        auto signedBlob = shared_model::crypto::DefaultCryptoAlgorithmType::sign(
-            shared_model::crypto::Blob(unsigned_.payload()), keypair);
-        iroha::protocol::Signature protosig;
-        protosig.set_pubkey(crypto::toBinaryString(keypair.publicKey()));
-        protosig.set_signature(crypto::toBinaryString(signedBlob));
-        auto *s1 = new Signature(protosig);
-        unsigned_.addSignature(detail::PolymorphicWrapper<Signature>(
-            s1));  // TODO: 05.12.2017 luckychess think about false case
+        auto signedBlob =
+            shared_model::crypto::DefaultCryptoAlgorithmType::sign(
```

Comment:
There is should used a crypto provider with template algorithm type:
```
shared_model::crypto::CryptoSigner<T>::sign
```
where `T` is a parameter of the method.